base/action.py

```diff
@@ -0,0 +1,7 @@
+from abc import ABC, abstractmethod
+
+
+class BaseAction(ABC):  # pylint: disable=too-few-public-methods
+    @abstractmethod
+    def perform_as(self):
```

Comment:
این perform_as از لحاظ انگلیسی معنی و منظورش چیه؟

Reply:
The phrase “perform as” typically means to carry out a task or function in the manner of something or someone else. It’s often used to describe how well or poorly something or someone executes a task or role, compared to a certain standard or expectation.

For example, if we say “a device performs as expected,” it means the device is functioning or operating in the way that was anticipated. If we say “an actor is performing as Hamlet,” it means the actor is playing the role of Hamlet.

In essence, “perform as” is about doing a task, role, or function in a certain way or to a certain standard. It can be used in various contexts, including work (e.g., “perform as a team”), technology (e.g., “the software performs as designed”), arts (e.g., “perform as a character in a play”), and more.

Reply:
در واقع داره میگه داره این مورد اجرا یا پرفورم میشه به این عنوان، این من عینا از خود دیزاینش برداشتم که اینجوری تعریف میکردش.